src/main/webapp/app/exercises/shared/result/result.utils.ts

```diff
@@ -364,3 +364,23 @@ export function getFeedbackByTestCase(testCase: string, feedbacks?: Feedback[]):
     const i = feedbacks.findIndex((feedback) => feedback.text?.localeCompare(testCase) === 0);
     return i !== -1 ? feedbacks[i] : null;
 }
+
+/**
+ * Removes references from the {@link Participation}, {@link Submission}, and {@link Feedback} back to the result itself.
+ *
+ * @param result
+ */
+export function breakCircularResultBackReferences(result: Result) {
+    if (result.participation?.results) {
+        result.participation.results = [];
```

Comment:
I’d expect that usually `result.submission.participation` and `result.participation` are the same. Since for the former the list was set to `[]` before as well, I think keeping the same here makes sense and should not break other components.